### PR TITLE
UPSERT support DUPLICATE KEY INGORE,TEST PASSING

### PR DESCRIPTION
**Pull request by 10km**
update JSqlParserCC.jjt,add ON DUPLICATE IGNORE support for  UPSERT statement,
_ON DUPLICATE IGNORE_ caluse is for phoenix,see https://phoenix.apache.org/language/index.html#upsert_values
update UpsertTest.java ,add _testUpsertIgnore_ test method for _ON DUPLICATE IGNORE_

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -377,6 +377,79 @@
                 </executions>
             </plugin>
         </plugins>
+        <pluginManagement>
+        	<plugins>
+        		<!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence on the Maven build itself.-->
+        		<plugin>
+        			<groupId>org.eclipse.m2e</groupId>
```
Comment (wumpz) [inline pom.xml:384]: Sure this mapping needs to be in this repository. Couldn't this be externalized?

File: src/main/java/net/sf/jsqlparser/statement/upsert/Upsert.java [hunk 2]
```diff
@@ -174,6 +185,10 @@ public Upsert withUseDuplicate(boolean useDuplicate) {
         return this;
     }
 
+    public Upsert withModifierIgnore(boolean modifierIgnore) {
+        this.setModifierIgnore(modifierIgnore);
+        return this;
+    }
```
Comment (wumpz) [inline src/main/java/net/sf/jsqlparser/statement/upsert/Upsert.java:191]: Please ensure a complete source code format.

File: src/test/java/net/sf/jsqlparser/statement/upsert/UpsertTest.java [hunk 3]
```diff
@@ -68,7 +68,21 @@ public void testUpsertDuplicate() throws JSQLParserException {
         assertTrue(upsert.isUseDuplicate());
         assertEquals(statement, "" + upsert);
     }
-
+    @Test
+    public void testUpsertIgnore() throws JSQLParserException {
+        String statement = "UPSERT INTO TEST (ID, COUNTER) VALUES (123, 0) ON DUPLICATE KEY IGNORE";
+        Upsert upsert = (Upsert) parserManager.parse(new StringReader(statement));
```
Comment (wumpz) [inline src/test/java/net/sf/jsqlparser/statement/upsert/UpsertTest.java:74]: Introduce a deparser test as well using those assert that a statement can be parsed and deparsed methods from TestUtils.
